openpgp/keys.go

```diff
@@ -442,6 +452,13 @@ EachPacket:
 			current.Name = pkt.Id
 			current.UserId = pkt
 		case *packet.Signature:
+			if pkt.SigType == packet.SigTypeKeyRevocation {
+				// These revocations won't revoke UIDs (see
+				// SigTypeIdentityRevocation). Handle these first,
+				// because key might have revocation coming from
+				// another key (designated revoke).
+				revocations = append(revocations, pkt)
```

Comment:
This is moved from line 503, i.e. it used to only append to `revocations` when `pkt.IssuerKeyId != nil && *pkt.IssuerKeyId != e.PrimaryKey.KeyId` did not evaluate to true. Now it always appends to `revocations` but that check is now at line 581 (for 1st party revocation) and there is additional logic regarding 3rd party revocations.